lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -421,21 +696,48 @@ bool mlir::triton::preProcessLoopAndGetSchedule(
     scf::ForOp &forOp, int numStages, mlir::triton::PipeliningOption &options) {
   // 1. First collect "interesting" operations with a stage where to schedule
   // them. This gives a coarse scheduling for the loop.
-  SmallVector<LoadDotOperand> loads;
+  llvm::MapVector<Operation *, PipelineOpInfo> opToInfo;
   bool hasMMAV3 = false;
-  collectOpsToPipeline(forOp, loads, hasMMAV3);
-  if (loads.empty())
+  if (!collectOpsToPipeline(forOp, opToInfo, numStages, hasMMAV3))
     return false;
-  bool hasAsynCp = llvm::any_of(loads, [](LoadDotOperand &load) {
-    return !isLoadFromTensorPtr(load.load);
+  bool hasAsynCp = llvm::any_of(opToInfo, [](auto &pair) {
```

Comment:
@ThomasRaoux do we even need this now, as blocked pointers are no longer legal at this stage anymore?

Reply:
right we shouldn't get any block pointers